spec/javascripts/stimulus/bulk_form_controller_test.js

```diff
@@ -11,39 +11,99 @@ describe("BulkFormController", () => {
     application.register("bulk-form", bulk_form_controller);
   });
 
+  // Mock I18n. TODO: moved to a shared helper
+  beforeAll(() => {
+    const mockedT = jest.fn();
+    mockedT.mockImplementation((string, opts) => (string + ', ' + JSON.stringify(opts)));
+
+    global.I18n =  {
+      t: mockedT
+    };
+  })
+
+  // (jest still doesn't have aroundEach https://github.com/jestjs/jest/issues/4543 )
+  afterAll(() => {
+    delete global.I18n;
+  })
+
   beforeEach(() => {
     document.body.innerHTML = `
       <form data-controller="bulk-form">
-        <input id="input1" type="text" value="initial1">
-        <input id="input2" type="text" value="initial2">
+        <div id="modified_summary" data-bulk-form-target="modifiedSummary" data-translation-key="modified_summary"></div>
+        <div data-record-id="1">
+          <input id="input1a" type="text" value="initial1a">
+          <input id="input1b" type="text" value="initial1b">
+        </div>
+        <div data-record-id="2">
+          <input id="input2" type="text" value="initial2">
+        </div>
       </form>
     `;
   });
 
-  describe("#toggleModified", () => {
-    it("marks a changed element as modified", () => {
-      // const form = document.getElementsByTagName("form")[0];
-      const input1 = document.getElementById("input1");
+  describe("Modifying input values", () => {
+    // This is more of a behaviour spec. Jest doesn't have all the niceties of RSpec so lots of code
+    // would be repeated if these were broken into multiple examples. So it seems impractical to
+    // write individual unit tests.
+    it("counts modified fields and records", () => {
```

Comment:
For setup in Jest you can do things like : 
 ```javascript
describe("", () => {
  let modified_summary

  beforeEach(() => {
    modified_summary = document.getElementById("modified_summary");
  })

  it("does something", () => {
    ....
    expect(modified_summary.textContent).toBe('modified_summary, {"count":1}');
  })
})
```
and you can also create reusable function if you need to repeat some test, ie :  https://github.com/openfoodfoundation/openfoodnetwork/blob/00b7ae9600fdb26495be282a7a3f3daf5ddae323/spec/javascripts/stimulus/tabs_and_panels_controller_test.js#L15C3-L23C6 

I would prefer the test to be split up, but it shouldn't block merging.

Reply:
Thanks for the feedback 👍

Reply:
I would have really liked to test each feature separately here, but to do that more efficiently would require `let`, and perhaps even a helper to compile only the necessary HTML tags for each test (which is starting to form in my head..).

But for now I've moved the variables to a `beforeEach`.

Reply:
Actually I went ahead and split the `modified` class tests out.